app/static/src/app/store/run/mutations.ts

```diff
@@ -122,5 +129,36 @@ export const mutations: MutationTree<RunState> = {
         if (paramSet) {
             paramSet.hidden = !paramSet.hidden;
         }
+    },
+
+    [RunMutation.SaveParameterDisplayName](state: RunState, payload: ParameterSetNames) {
+        const isDuplicateDisplayName = state.parameterSets.find((set: ParameterSet) => {
+            return set.displayName === payload.newDisplayName;
+        });
+        const isSpecialSetName = payload.newDisplayName.match(/^ *set \d+ *$/gi);
+        const paramSet = state.parameterSets.find((set: ParameterSet) => set.name === payload.parameterSetName);
+        if (paramSet) {
+            const isParamSetName = payload.newDisplayName === paramSet.name;
+            const isParamSetDisplayName = payload.newDisplayName === paramSet.displayName;
```

Comment:
Presumably if this is true then we could return immediately as there's nothing to do - the requested new display name is the same as the current display name. (Though if you follow my suggestion of clearing the error in this mutation, I guess you'd need to do that in the case of `isParamSetDisplayName` too.)

Reply:
hmm yh we can only return immediately (after clearing error) if `isParamSetDisplayName`. if `isParamSetName`, we still need to update the `displayName`. Think it feels more consistent to just have the error cases and the else block can just update the name